W-18359572 policy violations gr (#888)
mulesoft/docs-gateway: pdk/1.4/modules/ROOT/pages/policies-pdk-configure-features-violations.adoc

= Reporting Policy Violations
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]
:imagesdir: ../assets/images

NOTE: For an example of reporting policy violations, see https://github.com/mulesoft/pdk-custom-policy-examples/blob/support/{template-policies-url-ver-var}/stream-payload/README.md[Stream Payload Policy].

PDK provides functions to report a policy violation. Reported policy violations appear in Anypoint Monitoring.

> A request can only have one active policy violation. If a previous policy reports a policy violation, generating a new policy violation overrides the previous violation. If multiple policies generate policy violations for a single request, only one policy violation appears in Anypoint Monitoring. 

Reviewer: ```suggestion
Only one policy violation can be active for a request at a time. When a new violation occurs, it replaces any existing violation. If multiple policies generate violations, only the most recent one appears in Anypoint Monitoring.
```
This paragraph is a bit repetitive, I've suggested a rewording. Please check if it remains accurate.

Reporting a policy violation doesn't reject the request. To reject requests, see xref:policies-pdk-configure-features-stop.adoc[].

> To view the current policy violation or generate a new policy violation, PDK provides these functions:

Reviewer: ```suggestion
To view the current policy violation or generate a new policy violation, use these PDK functions:
```

[source,Rust]
----
/// Returns the existing policy violation of the current request. Returns None if no violation exists.
pub fn policy_violation(&self) -> Option<PolicyViolation>;

/// Generates a new policy violation for the current request. If a violation was previously generated, the previous violation is overridden.
pub fn generate_policy_violation(&self);

/// Generates a new policy violation with the client application details for the current request. This overwrites any previous policy violations.
pub fn generate_policy_violation_for_client_app<T: Into<String>, K: Into<String>>(
   &self,
   client_name: T,
   client_id: K,
);
----

> To retrieve information about the current policy violation, PDK provides these functions:

Reviewer: ```suggestion
To retrieve information about the current policy violation, use these PDK functions:
```
